skyportal/models.py

```diff
@@ -72,57 +72,99 @@ def is_owned_by(self, user_or_token):
+    targets for data sharing requests. `Photometry` and `Spectra` shared with
+    a `Group` will be visible to all its members. `Group`s maintain specific
+    `Stream` permissions. In order for a `User` to join a `Group`, the `User`
+    must have access to all of the `Group`'s data `Stream`s.
+    """
+
+    name = sa.Column(sa.String, unique=True, nullable=False, doc='Name of the group.')
 
     streams = relationship(
         'Stream',
         secondary='group_streams',
         back_populates='groups',
         passive_deletes=True,
+        doc='Stream access required for a User to become a member of the Group.',
+    )
+    filters = relationship(
+        "Filter",
+        back_populates="group",
+        passive_deletes=True,
+        doc='All filters (not just active) associated with ' 'a group.',
     )
-    filters = relationship("Filter", back_populates="group", passive_deletes=True)
 
     users = relationship(
-        'User', secondary='group_users', back_populates='groups', passive_deletes=True
+        'User',
+        secondary='group_users',
+        back_populates='groups',
+        passive_deletes=True,
+        doc='The members of this group.',
     )
+
     group_users = relationship(
         'GroupUser',
         back_populates='group',
         cascade='save-update, merge, refresh-expire, expunge',
         passive_deletes=True,
+        doc='Elements of a join table mapping Users to Groups.',
     )
 
-    observing_runs = relationship('ObservingRun', back_populates='group')
+    observing_runs = relationship(
+        'ObservingRun',
+        back_populates='group',
+        doc='The observing runs associated with ' 'this group.',
+    )
     photometry = relationship(
         "Photometry",
         secondary="group_photometry",
         back_populates="groups",
         cascade="save-update, merge, refresh-expire, expunge",
         passive_deletes=True,
+        doc='The photometry visible to this group.',
     )
+
     spectra = relationship(
         "Spectrum",
         secondary="group_spectra",
         back_populates="groups",
         cascade="save-update, merge, refresh-expire, expunge",
         passive_deletes=True,
+        doc='The spectra visible to this group.',
+    )
+    single_user_group = sa.Column(
+        sa.Boolean,
+        default=False,
+        doc='Flag indicating whether this group '
+        'is a singleton group for one user only.',
     )
-    single_user_group = sa.Column(sa.Boolean, default=False)
 
 
 GroupUser = join_model('group_users', Group, User)
-GroupUser.admin = sa.Column(sa.Boolean, nullable=False, default=False)
+GroupUser.__doc__ = "Join table mapping `Group`s to `User`s."
+
+GroupUser.admin = sa.Column(
+    sa.Boolean,
+    nullable=False,
+    default=False,
+    doc="Boolean flag indicating whether the User " "is an admin of the group.",
+)
 
 
 class Stream(Base):
-    name = sa.Column(sa.String, unique=True, nullable=False)
+    """A stream producing """
```

Comment:
Sounds like this was intended to be continued

Reply:
fixed